unikube/cli/init.py

```diff
@@ -0,0 +1,228 @@
+from typing import List, Optional, Tuple
+
+import click
+import yaml
+from pydantic import BaseModel
+
+from unikube.cli import console
+from unikube.cli.console import confirm, deck_list, organization_list, project_list
+
+
+class UnikubeFileBuild(BaseModel):
+    context: str = "."
+    docker_file: str = "Dockerfile"
+    target: str = None
+
+
+class UnikubeFileContext(BaseModel):
+    organization: str
+    project: str
+    deck: str
+
+
+class UnikubeFileApp(BaseModel):
+    context: UnikubeFileContext
+    build: UnikubeFileBuild
+    deployment: str
+    port: int
+    command: Optional[str]
+    volumes: Optional[List[str]]
+    env: Optional[List[str]]
+
+
+class UnikubeDumper(yaml.SafeDumper):
+    """Custom dumper for nice list identation.
+
+    See https://stackoverflow.com/questions/25108581/python-yaml-dump-bad-indentation
+    """
+
+    def increase_indent(self, flow=False, indentless=False):
+        return super(UnikubeDumper, self).increase_indent(flow, False)
+
+
+def _generate_unikube_file(apps: List[Tuple[str, UnikubeFileApp]], to_file: bool = True):
+    result = {"version": "1", "apps": {}}
+    for app in apps:
+        result["apps"].update({app[0]: app[1].dict(exclude_unset=True)})
+
+    if to_file:
+        with open("unikube.yaml", "w") as unikube_file:
+            yaml.dump(result, unikube_file, Dumper=UnikubeDumper, sort_keys=False)
+    else:
+        console.echo(yaml.dump(result, Dumper=UnikubeDumper, sort_keys=False))
+
+    return True
+
+
+def prompt_headline(text: str):
+    return click.echo(click.style(text + "\n", bold=True, underline=True))
+
+
+def format_question(question, question_only=False):
+    if not question_only:
+        mark = click.style("? ", fg=(69, 208, 147), bold=True)
+    else:
+        mark = ""
+    question_ = click.style(question, bold=True)
+    return f"{mark}{question_}"
+
+
+def prompt_for_choice(question: str, list_func, ctx):
+    click.echo(format_question(question, question_only=True))
+    result = list_func(ctx)
+    return result
+
+
+def path_validator(input_string: str):
+    return len(input_string) == 0 or (isinstance(input_string, str) and ":" in input_string)
+
+
+def env_validator(input_string: str):
+    return len(input_string) == 0 or (isinstance(input_string, str) and "=" in input_string)
+
+
+def get_service_name():
+    return console.input("What's the name of the service?", mandatory=True)
```

Comment:
@SteinRobert 
This is just the top-level key and does not matter too much in most cases where there is only one service declared in the unikube file, I guess. Maybe we should restrict this key to some typical legal identifier pattern, or provide a default like `"app"` (or query this later and use the deployment name as default). Since `"Service"` is a term from the kubernetes biotope, users might be wondering if this needs to be something specific (requiring chart knowledge).

Reply:
Removed the input - we're setting `app` statically as service name for now.